src/coreclr/jit/importercalls.cpp

```diff
@@ -10177,6 +10226,44 @@ GenTree* Compiler::impMinMaxIntrinsic(CORINFO_METHOD_HANDLE method,
     }
 #endif // FEATURE_HW_INTRINSICS && TARGET_XARCH
 
+#ifdef TARGET_RISCV64
+    op2 = impPopStack().val;
+    op1 = impPopStack().val;
+
+    GenTree *op1Clone = nullptr, *op2Clone = nullptr;
+    if (!isNumber)
+    {
+        op1 = impCloneExpr(op1, &op1Clone, CHECK_SPILL_NONE, nullptr DEBUGARG("Clone op1 for Math.Min/Max non-Number"));
+        op2 = impCloneExpr(op2, &op2Clone, CHECK_SPILL_NONE, nullptr DEBUGARG("Clone op2 for Math.Min/Max non-Number"));
```

Comment:
`CHECK_SPILL_NONE` here is wrong. It will reorder side effects of expressions pushed on the IL stack.

The simplest way to do this correctly is to pop `op2`, then clone it with `CHECK_SPILL_ALL`, then pop `op1` and then  clone it with `CHECK_SPILL_ALL`.